### PR TITLE
Authorization

### PR DESCRIPTION
**Pull request by lucascomamala**
**Note to reviewer:**
For your convenience, the seeds file comes preloaded with an admin and a standard user. The credentials are the following:

Admin:
> email: admin@mail.com
> password: 123456

User:
> email: user@mail.com
> password: 123456

- Install CanCanCan in your project.
- Add a role column to the users table. Remember to use a migration for this.
- A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
  - For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
- A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
  - For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -30,6 +30,12 @@ def create
     end
   end
 
+  def destroy
+    @post = Post.find(params[:id])
+    @post.destroy!
+    redirect_to user_path(@post.author), notice: 'You deleted a post!'
+  end
+
   private
```
Comment (rloterh) [inline app/controllers/posts_controller.rb:42]: - [x] Good job so far, however, let's also ensure that CanCanCan is used for Post authorization in the _Post Controller_ so as to allow authorization of resources. We can achieve this by adding helper methods like the example below:

```ruby
  class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Felipe & Lucas, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🏁 🚦🏃‍♀️ ⛪ 🚜 🏃 🌴 🏫🏃‍♂️ 🌳 🏘️  

## Highlights 👍  
- A nice job so far implementing authorization with CanCanCan!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rloterh** [comment]
@lucascomamala 
Kindly ensure to fix the Robocop linter check error asap. Cheers!

**Comment by lucascomamala** [comment]
@rloterh Done! sorry about that

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 class CommentsController < ApplicationController
+  load_and_authorize_resource
```
Comment (rloterh) [inline app/controllers/comments_controller.rb]: - [x] [OPTIONAL]
Kindly note that we only intend to allow authorization of resources on _Post_, and hence it's redundant to include `load_and_authorize_resource` in the _Comment_ controller.

File: app/controllers/likes_controller.rb [hunk 3]
```diff
@@ -1,4 +1,6 @@
 class LikesController < ApplicationController
+  load_and_authorize_resource
```
Comment (rloterh) [inline app/controllers/likes_controller.rb]: - [x] [OPTIONAL]
Similarly, we only intend to allow authorization of resources on _Post_, and hence it's redundant to include `load_and_authorize_resource` in the _Like_ controller.

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Felipe & Lucas,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Good job for introducing an improvement to authorization aspects!

## Optional suggestions 🧰 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

_Kindly consider my inline comments marked **[OPTIONAL]** under this review_

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**